lib/xwax/timecoder.c

```diff
@@ -629,6 +639,59 @@ static void process_bitstream(struct timecoder *tc, signed int m)
 	  tc->valid_counter);
 }
 
+/*
+ * Compare the last quadrant we were in to the new one and return the
+ * correct displacement for the pitch filter model
+ */
+int direction_change_counter = 0;
+static double quantize_phase(struct timecoder *tc)
+{
+    unsigned diff = ((tc->quadrant - tc->last_quadrant) % 4);
+
+    if (tc->direction_changed) {
+        direction_change_counter++;
+        debug("Direction changed %d times\n", direction_change_counter);
+    }
+
+    if (diff == 0 && !tc->direction_changed) { /* Check for a displacement of four quadrants */
+        debug("Skipped 3 quadrants\n");
+        return 1.0 / tc->def->resolution;
+
+    } else if ((tc->forwards && diff == 3) || (!tc->forwards && diff == 1) ) { /* Check for a displacement of three quadrants */
+        debug("Skipped 2 quadrants\n");
+        return (3.0 / tc->def->resolution) / 4.0;
```

Comment:
can you quickly explain where the $$n / \text{res} / m$$  formula and the values for `n`/`m` come from?